### PR TITLE
Upgrade to net8 and EF8

### PR DESCRIPTION
**Pull request by oleksii-korniienko**
**Describe the change**
Upgrade to .NET8 and use EF8

**Tests**
Same tests

**Breaking change**
Compatible with EF7 and net6

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -5,10 +5,10 @@
     <PackageLicenseUrl>https://github.com/SergiiKram/workflow-core/blob/idempotent-workflow-start/LICENSE.md</PackageLicenseUrl>
     <RepositoryType>git</RepositoryType>
     <RepositoryUrl>https://github.com/SergiiKram/workflow-core</RepositoryUrl>
-    <Version>3.9.0.4</Version>
-    <AssemblyVersion>3.9.0.4</AssemblyVersion>
-    <FileVersion>3.9.0.4</FileVersion>
+    <Version>4.0.0-alpha.1</Version>
+    <AssemblyVersion>4.0.0.0</AssemblyVersion>
+    <FileVersion>4.0.0.0</FileVersion>
```
Comment (SergiiKram) [inline src/Directory.Build.props]: Please don't update versions here, I do it before publishing and to match the original package versioning.

File: src/providers/WorkflowCore.Persistence.EntityFramework/WorkflowCore.Persistence.EntityFramework.csproj [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <AssemblyTitle>Workflow Core EntityFramework Core Persistence Provider</AssemblyTitle>
     <Authors>DanielGerlag,sergiikram</Authors>
-    <TargetFrameworks>netstandard2.1;net6.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.1;net6.0;net8.0</TargetFrameworks>
```
Comment (SergiiKram) [inline src/providers/WorkflowCore.Persistence.EntityFramework/WorkflowCore.Persistence.EntityFramework.csproj:6]: Please update all published projects to .NET 8:

image

**Review by SergiiKram — CHANGES_REQUESTED** [review]

File: src/providers/WorkflowCore.Persistence.EntityFramework/WorkflowCore.Persistence.EntityFramework.csproj (hunk 2 above)
Comment (oleksii-korniienko) [reply src/providers/WorkflowCore.Persistence.EntityFramework/WorkflowCore.Persistence.EntityFramework.csproj:6]: I updated only those projects that target .net6.
WorkflowCore, WorkflowCore.Providers.Redis, WorkflowCore.DSL, WorkflowCore.Testing targets netstandard2.0. I think we don't need to make them framework-specific. What do you think?

Comment (SergiiKram) [reply src/providers/WorkflowCore.Persistence.EntityFramework/WorkflowCore.Persistence.EntityFramework.csproj:6]: Ahh, I thought they already target net6. 
So then ok, no need to update them.

File: src/Directory.Build.props (hunk 1 above)
Comment (oleksii-korniienko) [reply src/Directory.Build.props]: reverted

**Review by SergiiKram — APPROVED** [review]

**Comment by SergiiKram** [comment]
Published 3.9.0.5

**Pull request merged**